### PR TITLE
Fix test kill container

### PR DESCRIPTION
**Pull request by brybacki**
**What this PR does / why we need it**:
Fix flaky test

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
#1285 
#1301

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by brybacki** [comment]
trying to reproduce error, it happened more often on crio lane

/test pull-containerized-data-importer-e2e-os-3.11.0-crio

**Comment by brybacki** [comment]
it works so good now, interesting
/test pull-containerized-data-importer-e2e-os-3.11.0-crio

**Comment by brybacki** [comment]
one more try:
/retest

**Comment by maya-r** [comment]
what does the exit code of 137 mean?

also: any change should probably be to the upload test as well.
#1301 says the upload test which also does "kill 1" is flaky as well.

**Comment by maya-r** [comment]
Sorry for the conflict. I was just looking at the same code and thought I should clean it up instead of pointing out it could be :-)
It seems like this is a scenario of the "kill 1" process being killed before it returns, or something. Sounds good to me.

**Comment by brybacki** [comment]
> It seems like this is a scenario of the "kill 1" process being killed before it returns, or something. Sounds good to me.

yes, and true this is used in a few places. But this is so rare that it is hard to catch and test. Now as I have seen failure in 3997, I'll update test cases and check a few more times

File: tests/cloner_test.go [hunk 1]
```diff
@@ -866,9 +867,12 @@ var _ = Describe("[rfe_id:1277][crit:high][vendor:cnv-qe@redhat.com][level:compo
...
 		By("Kill upload pod to force error")
-		_, errLog, err := f.ExecShellInPodWithFullOutput(targetNs.Name, utils.UploadPodName(targetPvc), "kill 1")
-		Expect(err).To(BeNil())
-		Expect(errLog).To(BeEmpty())
+		_, _, err = f.ExecShellInPod(utils.UploadPodName(targetPvc), targetNs.Name, "kill 1")
+		Expect(err).To(Or(
+			BeNil(),
+			WithTransform(errAsString, ContainSubstring("137"))))
+		// following log is needed just for an experiment, to be removed later.
+		fmt.Fprintf(GinkgoWriter, "INFO: err %v\n", err)
```
Comment (mhenriks) [inline tests/cloner_test.go:875]: removing this?

File: tests/upload_test.go [hunk 2]
```diff
@@ -674,10 +675,12 @@ var _ = Describe("[rfe_id:138][crit:high][vendor:cnv-qe@redhat.com][level:compon
...
-		By("Kill upload container to force restart")
-		_, errLog, err := f.ExecShellInPodWithFullOutput(f.Namespace.Name, utils.UploadPodName(pvc), "kill 1")
-		Expect(err).To(BeNil())
-		Expect(errLog).To(BeEmpty())
+		By("Kill upload pod to force error")
+		_, _, err = f.ExecShellInPod(utils.UploadPodName(pvc), f.Namespace.Name, "kill 1")
+		Expect(err).To(Or(
+			BeNil(),
+			WithTransform(errAsString, ContainSubstring("137"))))
+		fmt.Fprintf(GinkgoWriter, "INFO: err %v\n", err)
```
Comment (mhenriks) [inline tests/upload_test.go:683]: and this, I guess?

File: tests/cloner_test.go (hunk 1 above)
Comment (brybacki) [reply tests/cloner_test.go:875]: this is actually part of experiment I want to see the problem happen, and then I'll be sure that it works, 

but after confirming I can remove it

Comment (brybacki) [reply tests/cloner_test.go:875]: so far it is not failing often and mainly on crio, so It is hard to confirm, 
also it is not my priority right now so I am not trying too hard

**Comment by brybacki** [comment]
All the comments addressed, tests are green ;)
@awels @mhenriks what do you think?

**Comment by brybacki** [comment]
Finally, it proves that it is good enough for the test, all happened as expected 

https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/1297/pull-containerized-data-importer-e2e-os-3.11.0-crio/1285865764250718208/build-log.txt?Expires=1595411316&GoogleAccessId=prow-46%40kubevirt-240711.iam.gserviceaccount.com&Signature=NmYdQKYdovHij73IQ0DKTcpQAwUpPRZyi1tKSqpnqIB9lwlU%2BiqPzdlnqUGNFNJYBRb01N5Cw7dNNFapEGZZGf5JsFp0SSC9aFUjj6%2F6kxjAEij4dWsNh%2Fyt93Q5kRQ7Q8EaiWxjDuNxwc1XDGTrRl6hGWHYCtbuZ2Qdkgt9YYLCTND0WVRFJn5gezFhCKO4nRgnM6Okv1uUJgau5ZvmmCzqaf7fDZwtzgl6ZyDyxS0QsOtUgTXh4s6ZBxKYFoJSFwHMOUlkM7m%2Bs9kduaE4qjgrg2I2I2VRbFI%2BcldW%2FvDtnKCdkSSaJmmv4s57MUpmkP4wGn%2F%2Ff1bKn9UEg2O%2FMA%3D%3D

File: tests/cloner_test.go (hunk 1 above)
Comment (brybacki) [reply tests/cloner_test.go]: experiment finshed, this is now removed

File: tests/upload_test.go (hunk 2 above)
Comment (brybacki) [reply tests/upload_test.go]: the same here -removed

**Pull request merged**